pkg/agent/openflow/client_test.go

```diff
@@ -258,29 +268,986 @@ func TestConcurrentFlowInstallation(t *testing.T) {
 			var wg sync.WaitGroup
 			done := make(chan struct{})
 
-			for i := 0; i < 2; i++ {
-				wg.Add(1)
-				cacheKey := fmt.Sprintf(tc.cacheKeyFormat, i)
-				go func() {
-					defer wg.Done()
-					_, _ = tc.fn(ofClient, cacheKey) // in mock we trust
-				}()
-			}
-			go func() {
-				defer close(done)
-				wg.Wait()
-			}()
+			for i := 0; i < 2; i++ {
+				wg.Add(1)
+				cacheKey := fmt.Sprintf(tc.cacheKeyFormat, i)
+				go func() {
+					defer wg.Done()
+					_, _ = tc.fn(fc, cacheKey) // in mock we trust
+				}()
+			}
+			go func() {
+				defer close(done)
+				wg.Wait()
+			}()
+
+			select {
+			case <-time.After(time.Second):
+				close(timeoutCh)
+				t.Fatal("timeoutCh, maybe there are some deadlocks")
+			case <-done:
+				assert.True(t, concurrentCalls.Load().(bool))
+			}
+		})
+	}
+
+}
+
+func newFakeClient(mockOFEntryOperations *oftest.MockOFEntryOperations,
+	enableIPv4,
+	enableIPv6 bool,
+	nodeType config.NodeType,
+	trafficEncapMode config.TrafficEncapModeType,
+	options ...clientOptionsFn) *client {
+
+	o := &clientOptions{
+		enableProxy:           true,
+		enableAntreaPolicy:    true,
+		enableEgress:          true,
+		proxyAll:              true,
+		connectUplinkToBridge: false,
+		enableMulticast:       false,
+		enableTrafficControl:  false,
+		enableMulticluster:    false,
+	}
+	for _, fn := range options {
+		fn(o)
+	}
+
+	cli := NewClient(bridgeName,
+		bridgeMgmtAddr,
+		o.enableProxy,
+		o.enableAntreaPolicy,
+		o.enableEgress,
+		false,
+		o.proxyAll,
+		o.connectUplinkToBridge,
+		o.enableMulticast,
+		o.enableTrafficControl,
+		o.enableMulticluster)
+	client := cli.(*client)
+
+	var gwIPv4, gwIPv6 net.IP
+	gwMAC, _ := net.ParseMAC("00:00:10:10:00:01")
+	uplinkMAC, _ := net.ParseMAC("19:21:16:87:71:00")
```

Comment:
better declare them as global variables with some special meaning names and values that can be easily recognized, even though they are not used in the following test, to make developers easier to understand the flow.
```
// fakeGatewayMac is the MAC address of antrea-gw0 used in unit test in this file.
fakeGatewayMac _ = net.ParseMAC("0a:00:00:00:00:01")

// fakeUplinkMac is the MAC address of uplink device used in unit test in this file.
fakeUplinkMac _ = net.ParseMAC("0a:00:00:00:00:02")
```

Reply:
Changed it to global variables, but I keep the original MAC value since if I change the MAC value I need to change all related expected flows.

Reply:
Isn't it just a `shift+alt+R` operation? I think a fake value that could be easily recogized is helpful for readability when it's not convinient to reference the variables in expected flows. Otherwise it's hard to remember which MAC is what when looking at the tests, especially when more MAC addresses are used.

Reply:
Done.